### PR TITLE
review: fix: skip synthetic break from JDT

### LINKED ISSUE #5743: [Bug]: Spoon inserts incorrect break in enhanced switch block
### Describe the bug

As seen in the outputs below, spoon inserts a break statement at the end of blocks in an enhanced (arrow) switch that don't exist in the original source code
Note: With complianceVersion < 14 the bug doesn't appear. But those versions should not support enhanced switches at all, so...

### Source code you are trying to analyze/transform

```Java
class Main {
    void main() {
        switch(0) {
            case 1 -> {
            }
        }
    }
}
```

### Source code for your Spoon processing

```Java
import spoon.Launcher;
import spoon.SpoonException;
import spoon.reflect.declaration.CtClass;
import spoon.reflect.declaration.CtType;
import spoon.support.compiler.VirtualFile;

import java.util.Collection;

public class Main {

    public static void main(String[] args) {
        CtClass<?> clazz = parseClass("""
                class Main {
                    void main() {
                        switch(0) {
                            case 1 -> {
                            }
                        }
                    }
                }
                """);
        System.out.println(clazz);
    }

    public static CtClass<?> parseClass(String code) {
        Launcher launcher = new Launcher();
        launcher.addInputResource(new VirtualFile(code));
        launcher.getEnvironment().setNoClasspath(true);
        launcher.getEnvironment().setAutoImports(true);
        launcher.getEnvironment().setComplianceLevel(21);
        Collection<CtType<?>> allTypes = launcher.buildModel().getAllTypes();
        try {
            return (CtClass<?>) allTypes.stream().findFirst().get();
        } catch (ClassCastException e) {
            throw new SpoonException("parseClass only considers classes (and not interfaces and enums). Please consider using a Launcher object for more advanced usage.");
        }
    }
}
```

### Actual output

```Java
class Main {
    void main() {
        switch (0) {
            case 1 ->
                {
                    break;
                }
        }
    }
}
```

### Expected output

```Java
class Main {
    void main() {
        switch (0) {
            case 1 ->
                {
                }
        }
    }
}
```

### Spoon Version

11.0.0

### JVM Version

21

### What operating system are you using?

Linux

### PR DESCRIPTION
**Pull request by SirYwell**
Fixes #5743

### CODE REVIEW

File: src/main/java/spoon/support/compiler/jdt/JDTTreeBuilder.java [hunk 1]
```diff
@@ -342,6 +342,9 @@ public void endVisit(Block block, BlockScope scope) {
 
 	@Override
 	public void endVisit(BreakStatement breakStatement, BlockScope scope) {
+		if (breakStatement.isSynthetic) {
+			return; // we never entered
```
Comment (I-Al-Istannen) [inline src/main/java/spoon/support/compiler/jdt/JDTTreeBuilder.java:346]: Why handle this then instead of throwing an exception?

Comment (SirYwell) [reply src/main/java/spoon/support/compiler/jdt/JDTTreeBuilder.java:346]: This method is called by JDT, whether we want or not (i.e. even if we return false in visit(BreakStatement, ...))

Comment (I-Al-Istannen) [reply src/main/java/spoon/support/compiler/jdt/JDTTreeBuilder.java:346]: ah right we would need to return false in the *parent* construct. That makes sense

**Comment by MartinWitt** [comment]
Thanks @SirYwell

**Pull request merged**